app/src/ui/check-runs/ci-check-run-actions-logs.tsx

```diff
@@ -49,23 +61,176 @@ export class CICheckRunActionLogs extends React.PureComponent<
     }
   }
 
-  public render() {
-    const { output, loadingLogs } = this.props
+  private calculateLineLogNumberWidth(
+    logLinesData: ReadonlyArray<ILogLineTemplateData>
+  ): void {
+    if (logLinesData.length === 0) {
+      this.logLineNumberWidth = MIN_LOG_LINE_NUMBER_WIDTH
+      return
+    }
 
-    if (loadingLogs) {
-      return <>Loading…</>
+    const lastLine = logLinesData[logLinesData.length - 1]
+    const calcWidth =
+      lastLine.lineNumber.toString().length *
+      INDIVIDUAL_LOG_LINE_NUMBER_WIDTH_ALLOWED
+
+    this.logLineNumberWidth = Math.min(
+      MAX_LOG_LINE_NUMBER_WIDTH,
+      Math.max(MIN_LOG_LINE_NUMBER_WIDTH, calcWidth)
+    )
+  }
+
+  private renderLogs(log: string | undefined): JSX.Element | null {
+    if (log === undefined) {
+      return null
     }
 
-    if (output.type !== RefCheckOutputType.Actions) {
-      // This shouldn't happen, should only be provided actions type
-      return <>Unable to load logs.</>
+    const logParser = new ActionsLogParser(log, '')
+    const logLinesData = logParser.getParsedLogLinesTemplateData()
+    this.calculateLineLogNumberWidth(logLinesData)
+
+    const logLineJSX: JSX.Element[] = []
+
+    let logGroup: ILogLineTemplateData[] = []
+    for (let i = 0; i < logLinesData.length; i++) {
+      const lineData = logLinesData[i]
+      const isNextLineDataInGroup = logLinesData[i + 1]?.inGroup
```

Comment:
Ultra minor nit: if we want to avoid using `!` for falsey/truthy conditions, we should provide a default `false` value for this:
```suggestion
      const isNextLineDataInGroup = logLinesData[i + 1]?.inGroup ?? false
```

Feel free to ignore it if you want 😅

Reply:
:) I will go ahead and merge but catch it in another PR. :